applications/ConstitutiveLawsApplication/custom_processes/advance_in_time_high_cycle_fatigue_process.cpp

```diff
@@ -0,0 +1,271 @@
+    std::vector<double> rev_factor_rel_error;
+    std::vector<double> s_th;
+    std::vector<double> max_stress;
+
+    double acumulated_max_stress_rel_error = 0.0;
+    double acumulated_rev_factor_rel_error = 0.0;
+    bool fatigue_in_course = false;
+
+    const auto& r_elem = mrModelPart.ElementsBegin();
+    const unsigned int number_of_ip = r_elem->GetGeometry().IntegrationPoints(r_elem->GetIntegrationMethod()).size();
+
+    for (auto& r_elem : mrModelPart.Elements()) {   //This loop is done for all the integration points even if the cycle has not changed
+                                                    //in order to guarantee the stable condition in the WHOLE model (when Smax > Sth)
+        r_elem.Id();
+        r_elem.CalculateOnIntegrationPoints(MAX_STRESS_RELATIVE_ERROR, max_stress_rel_error, process_info);
+        r_elem.CalculateOnIntegrationPoints(REVERSION_FACTOR_RELATIVE_ERROR, rev_factor_rel_error, process_info);
+        r_elem.CalculateOnIntegrationPoints(THRESHOLD_STRESS, s_th, process_info);
+        r_elem.CalculateOnIntegrationPoints(MAX_STRESS, max_stress, process_info);
+
+        for (unsigned int i = 0; i < number_of_ip; i++){
+            if (max_stress[i] > s_th[i]) {
+                fatigue_in_course = true;
+                acumulated_max_stress_rel_error += max_stress_rel_error[i];
+                acumulated_rev_factor_rel_error += rev_factor_rel_error[i];
+            }
+        }
+    }
+    if ((acumulated_max_stress_rel_error < 1e-4 && acumulated_rev_factor_rel_error < 1e-4 && fatigue_in_course) || (DamageIndicator && acumulated_max_stress_rel_error < 1e-3 && acumulated_rev_factor_rel_error < 1e-3 && fatigue_in_course)) {
+        rAdvancingStrategy = true;
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void AdvanceInTimeHighCycleFatigueProcess::TimeIncrement(double& rIncrement)
+{
+    auto& process_info = mrModelPart.GetProcessInfo();
+    std::vector<double>  cycles_to_failure_element;
+    std::vector<int>  local_number_of_cycles;
+    std::vector<double> period;
+    std::vector<double> s_th;
+    std::vector<double> max_stress;
+    double min_time_increment;
+    double time = process_info[TIME];
+    bool current_constraints_process_list_detected = false;
+
+    double user_avancing_time = mThisParameters["fatigue"]["advancing_strategy_time"].GetDouble();
+    double user_avancing_cycles = mThisParameters["fatigue"]["advancing_strategy_cycles"].GetDouble();
+    std::vector<std::string> constraints_list = mThisParameters["fatigue"]["constraints_process_list"].GetStringArray();
+    double model_part_final_time = mThisParameters["problem_data"]["end_time"].GetDouble();
+    for (unsigned int i = 0; i < constraints_list.size(); i++){
+        for (unsigned int j = 0; j < mThisParameters["processes"]["constraints_process_list"].size(); j++){

```

Comment:
To correctly perform the calculation associated to the advance in time process I need to know exactly which is the model part responsible of the fatigue phenomena, thus I need the information that I obtain through the "constraint_process_list" argument. If there is any other way to access to this information I can adapt it but I am not familiar with it so if you can suggest something it would really help! Thx

Reply:
Ah ok now I see
I recommend making this an input for the process, i.e. explicitly tell the process which ModelParts are responsible

Reply:
I've discussed this with @AlejandroCornejo and the main issue is that I require the Interval information and so this is the easiest way to access this information. I feel like if sometime this become deprecated I could think in modifying it but I would suggest to leave it like it is for this PR.